src/openklant/components/klantinteracties/api/tests/test_klantcontacten.py

```diff
@@ -1652,19 +1660,272 @@ def test_filter_onderwerpobject_by_klantcontact_filters(self):
                 self.assertEqual(len(data["results"]), 1)
                 self.assertEqual(data["results"][0]["uuid"], str(obj.uuid))
 
-    def test_destroy_onderwerpobject(self):
-        onderwerpobject = OnderwerpobjectFactory.create()
-        detail_url = reverse(
+    def test_destroy_with_cascade_true(self):
+        klantcontact = KlantcontactFactory.create()
+        was_klantcontact = KlantcontactFactory.create()
+
+        onderwerpobject = OnderwerpobjectFactory.create(
+            klantcontact=klantcontact,
+            was_klantcontact=was_klantcontact,
+        )
+
+        url = reverse(
             "klantinteracties:onderwerpobject-detail",
-            kwargs={"uuid": str(onderwerpobject.uuid)},
+            kwargs={"uuid": onderwerpobject.uuid},
         )
-        response = self.client.delete(detail_url)
+
+        response = self.client.delete(f"{url}?cascade=true")
+
         self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
 
-        list_url = reverse("klantinteracties:onderwerpobject-list")
-        response = self.client.get(list_url)
-        data = response.json()
-        self.assertEqual(data["count"], 0)
+        self.assertFalse(
+            Onderwerpobject.objects.filter(uuid=onderwerpobject.uuid).exists()
+        )
+
+        self.assertFalse(Klantcontact.objects.filter(uuid=klantcontact.uuid).exists())
+        self.assertFalse(
+            Klantcontact.objects.filter(uuid=was_klantcontact.uuid).exists()
+        )
+
+    def test_destroy_with_cascade_false(self):
+        klantcontact = KlantcontactFactory.create()
+        onderwerpobject = OnderwerpobjectFactory.create(klantcontact=klantcontact)
+
+        url = reverse(
+            "klantinteracties:onderwerpobject-detail",
+            kwargs={"uuid": onderwerpobject.uuid},
+        )
+
+        response = self.client.delete(f"{url}?cascade=false")
+
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT)
+
+        self.assertFalse(
+            Onderwerpobject.objects.filter(uuid=onderwerpobject.uuid).exists()
+        )
+
+        self.assertTrue(Klantcontact.objects.filter(uuid=klantcontact.uuid).exists())
+
+    def test_destroy_with_cascade_shared_klantcontact(self):
+        shared_klantcontact = KlantcontactFactory.create()
+        shared_was_klantcontact = KlantcontactFactory.create()
+
+        onderwerpobject1 = OnderwerpobjectFactory.create(
+            klantcontact=shared_klantcontact,
+            was_klantcontact=shared_was_klantcontact,
+        )
+        onderwerpobject2 = OnderwerpobjectFactory.create(
+            klantcontact=shared_klantcontact,
+            was_klantcontact=shared_was_klantcontact,
+        )
+
+        url = reverse(
+            "klantinteracties:onderwerpobject-detail",
+            kwargs={"uuid": onderwerpobject1.uuid},
+        )
+
+        response = self.client.delete(f"{url}?cascade=true")
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Let's also add an assertion here on the response to verify that it contains the list of remaining klantcontact URLs